### ISSUE TITLE
Cluster autodetection did not find any active node. Make sure a GET /_nodes reguest on the hosts defined in the config returns the "http_address" field for each node.

### ISSUE THREAD

**Issue by harryheng** [yiisoft/yii2-elasticsearch#177]
我在、/common/config/main-local.php中components中已加入  'elasticsearch' => [
            'class' => 'yii\elasticsearch\Connection',
            'nodes' => [
                ['http_address' => '127.0.0.1:9200'],
                // configure more hosts if you have a cluster
            ],
        ],
还是报错怎么解决？

**Comment by tarunjangra** [comment]
I think earlier version of ES were returning http_address with API  **es/_nodes**.  ES 6.x is not returning http_address in payload.

**Comment by tarunjangra** [comment]
http_address has been removed since elasticsearh 5.x in this pull request [#19218](https://github.com/elastic/elasticsearch/pull/19218). 

For the quick solution. put following line in connection.php at 187 line. 

```php
$node['http_address'] = $node['http']['publish_address'];
```

Just make sure it is above the following code.
 
```php
 if (!isset($node['http_address'])) {
     unset($nodes[$key]);
 }
```

**Comment by tarunjangra** [comment]
BTW, there is 2.1-dev version available in yiisoft/yii2-elasticsearch. I've seen, this particular problem is solved there.

**Comment by Chienle** [comment]
Hey guys,
Today I've got some above issue with Elasticsearch below:
Cluster autodetection did not find any active node. Make sure a GET /_nodes reguest on the hosts defined in the config returns the "http_address" field for each node.

Ho to fix it ?

**Comment by nickzws** [comment]
put following line in connection.php at 187 line.

$node['http_address'] = $node['http']['publish_address'];


my elasticsearch is V6.3.0
 it is don`t working still!

**Comment by githubjeka** [comment]
https://github.com/olivere/elastic/commit/2c148092bf74bb366a97602e8ebd2d14524e5e7e

> Elasticsearch 5.0 removed the `http_address` and `https_address` from
the Nodes Info API. We'll use the `http` section of the API now.

REF #137 

Need fix PR or use `"yiisoft/yii2-elasticsearch": "2.0.4",`

**Comment by cebe** [comment]
@githubjeka could you check if this is fixed on the 2.1 branch?

all: see https://github.com/yiisoft/yii2-elasticsearch#requirements for compatilibilty information of extension version and eleasticsearch version.

**Comment by mikehaertl** [comment]
@cebe Is there any chance to get this fixed maybe in a bugfix release? This bug is a main blocker for the whole extension. It would require you to use a very old Elasticsearch version. The fix is easy as noted above. 

2.1 branch doesn't help much as there's no release either.

**Comment by machour** [comment]
@mikehaertl did you test the 2.1 branch nonetheless? If you report that it work, this will help get it released.

Don't hesitate to open a pull request with the mentioned fix for the 2.0 branch too, to help things move forward.

**Comment by samdark** [comment]
2.1 works at yiiframework.com: https://github.com/yiisoft-contrib/yiiframework.com/blob/master/composer.json#L21

What we need is to fix build for 2.0, release last version of it then do the same for 2.1.

**Comment by mikehaertl** [comment]
@samdark I think the fix for 2.0 release is not so easy as I thought. I've tried with disabling auto-detection and get yet another error now.

```
Elasticsearch request failed with code 400.
Error Info: Array
(
    [requestMethod] => GET
    [requestUrl] => http://elastic:9200//_cat/health?v
    [requestBody] => 
    [responseCode] => 400
    [responseHeaders] => Array
        (
            [access-control-allow-credentials] => true
            [content-type] => application/json; charset=UTF-8
            [content-length] => 257
        )

    [responseBody] => Array
        (
            [error] => Array
                (
                    [root_cause] => Array
                        (
                            [0] => Array
                                (
                                    [type] => illegal_argument_exception
                                    [reason] => request [//_cat/health] contains unrecognized parameter: [v]
                                )

                        )

                    [type] => illegal_argument_exception
                    [reason] => request [//_cat/health] contains unrecognized parameter: [v]
                )

            [status] => 400
        )

)
```

Problems seems to be the duplicate slash `//_cat/health`. The same request with single slash works fine with curl. I don't have this slash in my `nodes` config.

**Comment by mikehaertl** [comment]
Never mind - the problem was in my code: 

```
Yii::$app->elastic->get('/_cat/health?v');
```

Tried with 2.1 now and at least can get a response if i use $raw = true with get(). (without $raw requesting e.g. `_nodes` still fails with an "Response content must not be an array." error - whatever that means).

**Comment by beowulfenator** [comment]
Fixed in 2.0.4 (and now merged into master along with 2.1)
